src/main/java/ro/msg/learning/shop/model/Location.java

```diff
@@ -27,4 +29,21 @@ public void setStocks(Set<Stock> stocks) {
         this.stocks = stocks;
         stocks.forEach(stock -> stock.setLocation(this));
     }
+
+    public void removeProductFromStock(@NonNull Product product, @NonNull Integer quantity) {
```

Comment:
Please keep the model entities as POJO (no implementation inside the entities). This method should be part of a service class.